tests/dualtor/test_tunnel_memory_leak.py

```diff
@@ -0,0 +1,183 @@
+    operation. Since tunnel_packet_handler is only triggered by the
+    first packet, loop the process for all severs to trigger it as much
+    as possible.
+    """
+    @contextlib.contextmanager
+    def prepare_services(ptfhost):
+        """
+        Temporarily stop garp service, start arp and icmp service.
+        It has to stop garp_service for triggering tunnel_packet_handler.
+        It has to start arp and icmp service for receiving packets at server side. 
+        """
+        ptfhost.shell("supervisorctl stop garp_service")
+        ptfhost.shell("supervisorctl start arp_responder")
+        ptfhost.shell("supervisorctl start icmp_responder")
+        yield
+        ptfhost.shell("supervisorctl start garp_service")
+        ptfhost.shell("supervisorctl stop arp_responder")
+        ptfhost.shell("supervisorctl stop icmp_responder")
+	
+    @contextlib.contextmanager
+    def remove_neighbor(duthost, server_ip):
+        """
+        Remove ip neighbor before test for triggering tunnel_packet_handler,
+        restore it after test
+        """
+        flush_neighbor_ct = flush_neighbor(duthost, server_ip, True)
+
+        with flush_neighbor_ct:
+            command = "ip neighbor show %s" % server_ip
+            output = [_.strip() for _ in duthost.shell(command)["stdout_lines"]]
+            pytest_assert(not output, "server ip {} isn't flushed in neighbor table.".format(server_ip))
+            yield
+
+    pytest_assert(is_tunnel_packet_handler_running(upper_tor_host), 
+                "tunnel_packet_handler is not running in SWSS conainter.")
+
+    ptf_t1_intf = random.choice(get_t1_ptf_ports(lower_tor_host, tbinfo))
+
+    all_servers_ips = mux_cable_server_ip(upper_tor_host)
+    # Get the original memeory percent before test.
+    check_memory_leak(upper_tor_host)
+    with prepare_services(ptfhost):
+        for iface, server_ips in all_servers_ips.items():
+            server_ipv4 = server_ips["server_ipv4"].split("/")[0]
+            logging.info("Select DUT interface {} and server IP {} to test.".format(iface, server_ipv4))
+
+            pkt, exp_pkt = build_packet_to_server(lower_tor_host, ptfadapter, server_ipv4)
+
+            rm_neighbor = remove_neighbor(upper_tor_host, server_ipv4)
+
+            server_traffic_monitor = ServerTrafficMonitor(
+                upper_tor_host, ptfhost, vmhost, tbinfo, iface,
+                conn_graph_facts, exp_pkt, existing=True, is_mocked=True
+            )
+            with rm_neighbor, server_traffic_monitor:
+                testutils.send(ptfadapter, int(ptf_t1_intf.strip("eth")), pkt, count=PACKET_COUNT)
+                logging.info("Sent {} packets from ptf t1 interface {} on standby TOR {}"
+                            .format(PACKET_COUNT, ptf_t1_intf, lower_tor_host.hostname))
+
+                if not validate_neighbor_entry_exist(upper_tor_host, server_ipv4):
+                    pytest.fail("The server ip {} doesn't exist in neighbor table on dut {}. \
```

Comment:
Might be cleaner to use `pytest_assert` here similar to below.

Reply:
Done